mypy/type_visitor.py

```diff
@@ -347,3 +349,121 @@ def query_types(self, types: Iterable[Type]) -> T:
                 self.seen_aliases.add(t)
             res.append(t.accept(self))
         return self.strategy(res)
+
+
+class TypeQueryBool(SyntheticTypeVisitor[bool]):
+    """Specialized visitor for boolean strategies
+
+    strategy is used to combine results for a series of types,
+    Cases involve a boolean query using `any` or `all`.
+
+    Note: this visitor keeps an internal state (tracks type aliases to avoid
+    recursion), so it should *never* be re-used for querying different types,
+    create a new visitor instance instead.
+
+    # TODO: check that we don't have existing violations of this rule.
+    """
+    STRATEGY_ANY = 0  # type: Final[int]
+    STRATEGY_ALL = 1  # type: Final[int]
+
+    def __init__(self, strategy: int) -> None:
+        # 0: any()
+        # 1: all()
+        self.strategy = strategy
+        # Keep track of the type aliases already visited. This is needed to avoid
+        # infinite recursion on types like A = Union[int, List[A]].
+        self.seen_aliases = set()  # type: Set[TypeAliasType]
+
+    def bool_strategy_empty(self) -> bool:
+        if self.strategy == self.STRATEGY_ALL:
```

Comment:
Try accessing the constants as `TypeQueryBool.STRATEGY_ALL` instead. mypyc should directly emit a constant in that case, and it'll emit an object lookup for `self`. (That's probably a bug.) The object lookup should be cheap, but is still a memory access and an error check, so the constant might be better.